sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/JsonExpressionsSuite.scala

```diff
@@ -791,4 +791,30 @@ class JsonExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper with
         checkDecimalInfer(_, """struct<d:decimal(7,3)>""")
     }
   }
+
+  test("Length of JSON array") {
+    val null_json_array = """"""
+    val simple_json_array = """[1,2,3]"""
+    val empty_json_array = """[]"""
+    val json_array_of_array = """[[1],[2,3],[]]"""
+    val json_array_of_objects = """[{"a":123},{"b":"hello"}]"""
+    val complex_json_array = """[1,2,3,[33,44],{"key":[2,3,4]}]"""
+    val not_a_json_array = """{"key":"not a json array"}"""
+    val invalid_json_array = """[1,2,3,4,5"""
+
+    checkEvaluation(LengthOfJsonArray(Literal(null_json_array)), null)
+    checkEvaluation(LengthOfJsonArray(Literal(simple_json_array)), 3)
+    checkEvaluation(LengthOfJsonArray(Literal(empty_json_array)), 0)
+    checkEvaluation(LengthOfJsonArray(Literal(json_array_of_array)), 3)
+    checkEvaluation(LengthOfJsonArray(Literal(json_array_of_objects)), 2)
+    checkEvaluation(LengthOfJsonArray(Literal(complex_json_array)), 5)
+    checkEvaluation(LengthOfJsonArray(Literal(invalid_json_array)), null)
+
+    val exception = intercept[TestFailedException]{
+      checkEvaluation(LengthOfJsonArray(Literal(not_a_json_array)), null)
+    }.getCause
+
+    assert(exception.isInstanceOf[IllegalArgumentException])
```

Comment:
Shall we remove this? You can use `val exception = intercept[IllegalArgumentException]{` at line 813.